test/CoreTypechecker.hs

```diff
@@ -204,14 +206,44 @@ kApp1 =
         ( IR.Pi
             (SNat 1)
             (IR.PrimTy Nat)
-            (IR.Pi (SNat 1) (IR.PrimTy Nat) (IR.PrimTy Nat))
+            (IR.Pi (SNat 0) (IR.PrimTy Nat) (IR.PrimTy Nat))
         )
     )
     (IR.Elim (IR.Prim (Natural 1)))
 
 natToNatTy ∷ NatAnnotation
 natToNatTy =
-  (SNat 1, IR.VPi (SNat 1) (IR.VPrimTy Nat) (const (IR.VPrimTy Nat)))
+  ( SNat 1,
+    IR.VPi (SNat 0) (IR.VPrimTy Nat) (const (IR.VPrimTy Nat))
+  )
+
+-- ((K: Nat -> (Nat -> Nat) -> Nat) 1) should type check to (Nat -> Nat) -> Nat
+kFunApp1 ∷ NatElim
+kFunApp1 =
+  IR.App
+    ( IR.Ann
+        (SNat 1)
+        kcombinator
+        ( IR.Pi
+            (SNat 1)
+            (IR.PrimTy Nat)
+            ( IR.Pi
+                (SNat 0)
+                (IR.Pi (SNat 1) (IR.PrimTy Nat) (IR.PrimTy Nat))
+                (IR.PrimTy Nat)
+            )
+        )
+    )
+    (IR.Elim (IR.Prim (Natural 1)))
+
+kFunApp1CompTy ∷ NatAnnotation
+kFunApp1CompTy =
+  ( SNat 1,
+    IR.VPi
+      (SNat 0)
+      (IR.VPi (SNat 1) (IR.VPrimTy Nat) (const (IR.VPrimTy Nat)))
+      (const (IR.VPrimTy Nat))
+  )
 
 --K: (Nat -> Nat) -> Nat -> (Nat -> Nat) I:Nat -> Nat type checks to Nat -> (Nat -> Nat)
 kAppI ∷ NatElim
```

Comment:
This looks good, but I was wondering if you think it's worth also adding another test where the argument (L265-271) is replaced with just an unannotated `identity`? Because I think the typechecker should be able to push in the function argument's type and still work.

Reply:
Good idea! Added. It did pass.